backend/src/core/models/game.spec.js

```diff
@@ -1,583 +1,779 @@
+        const game = getXGamesWithoutDetails(1)[0];
+        const page = getParsedHtmlPage(riskOfRainHtmlDetailsPageMissingInfo);
 
-      it("The result has a property developers, which equals an empty array", function () {
-        expect(this.result.developers).toEqual([]);
+        this.result = game.getDescription(page);
       });
 
-      it("The game has three new player history entries", function () {
-        expect(this.result.playerHistory.length).toBe(4);
+      it("the result is an empty string", function () {
+        expect(this.result).toEqual("");
       });
+    });
+
+    describe("if the provided HTML page includes a description,", function () {
+      beforeAll(function () {
+        const game = getXGamesWithoutDetails(1)[0];
+        const page = getParsedHtmlPage(mortalDarknessGameHtmlDetailsPage);
 
-      it("All player history entries are in correct order", function () {
-        expect(this.result.playerHistory[0].averagePlayers).toBe(55);
-        expect(this.result.playerHistory[1].averagePlayers).toBe(5);
-        expect(this.result.playerHistory[2].averagePlayers).toBe(15);
-        expect(this.result.playerHistory[3].averagePlayers).toBe(513);
+        this.result = game.getDescription(page);
       });
 
-      it("The games player history entries are new PlayerHistory class instances", function () {
-        expect(this.result.playerHistory[0]).toBeInstanceOf(PlayerHistory);
-        expect(this.result.playerHistory[1]).toBeInstanceOf(PlayerHistory);
-        expect(this.result.playerHistory[2]).toBeInstanceOf(PlayerHistory);
-        expect(this.result.playerHistory[3]).toBeInstanceOf(PlayerHistory);
+      it("the game's description is returned", function () {
+        expect(this.result).toBe(
+          "“One grim dawn and noble I wake, The darkness is rampant, our oath shall break. A noble warrior soon shall rise, and clear the air of the darkened skies.”",
+        );
       });
     });
   });
 
   describe(".updateGameDetails", function () {
-    describe("When we try to update missing developers,", function () {
+    describe("When we try to update a game's details", function () {
       beforeAll(function () {
-        this.developers = ["Valve", "Test Dev"];
+        this.game = getXGamesWithoutDetails(1)[0];
+
+        const page = getParsedHtmlPage(karmazooHtmlDetailsPageSteamDb);
+
+        this.game.updateGameDetails(page);
+      });
 
+      it("the game's details are updated", function () {
+        expect(this.game.developers).toEqual(["Pastagames"]);
+        expect(this.game.genres).toEqual(["Action", "Casual"]);
+        expect(this.game.description).toEqual(
+          "Share the love and die trying! KarmaZoo is a joyful, altruistic, cooperative platformer where up to 10 random players help each other with the unique abilities of 50 different characters - all for the sake of good Karma.",
+        );
+      });
+    });
+  });
+
+  describe(".updateDevelopers", function () {
```

Comment:
Redundant tests. If you use encapsulation as above you would only write tests against one method.